### PR TITLE
Add Objective-C example

### PR DESCRIPTION
**Pull request by bugaevc**
Implement #10

### CODE REVIEW

File: objc/main.m [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    NSAutoreleasePool *pool;
+    CGTKWindow *window;
+    CGTKHeaderBar *headerbar;
+    CGTKLabel *label;
+
+    /*
+     * Create an autorelease pool that will be used for
+     * all [object autorelease] operations.
+     */
+    pool = [NSAutoreleasePool new];
```
Comment (sdroege) [inline objc/main.m]: Would it be possible to use ARC instead? See also https://developer.apple.com/library/content/releasenotes/ObjectiveC/RN-TransitioningToARC/Introduction/Introduction.html which seems to suggest that manual reference counting (including NSAutoreleasePool) is to be deprecated

Comment (bugaevc) [inline objc/main.m]: I don't think GCC has support for ARC.

Comment (sdroege) [inline objc/main.m]: It doesn't, no. Unfortunately the GCC implementation of ObjC is basically years behind everything and quite incomplete.

Comment (bugaevc) [inline objc/main.m]: If you can tell me how to make objc (with blocks and all the modern features) work in clang on Linux without manually installing and compiling external runtimes...
E.g. to compile this I simply installed `gnustep*` and `gcc-objc` (on Fedora).

Comment (sdroege) [inline objc/main.m]: AFAIU you can't. Apple only added all these features to their gcc fork (and it was never merged back), and AFAIK nothing is in the free clang version. In addition to the compiler, you also need a up-to-date language runtime, etc.

So if your goal is to target ObjC on non-Apple platforms, this is the way to go. On Apple platforms it's all using old and deprecated API.

Comment (bugaevc) [inline objc/main.m]: Core GTK itself doesn't use ARC (and is not easily convertible), so let's leave this with manual MM.

By the way, we have full Objective-C support in Darling (https://darlinghq.org), so I believe all the necessary parts are free/open-source, but you have to compile a lot of components that distros don't ship.

Comment (sdroege) [inline objc/main.m]: Ah, good to know

**Comment by bugaevc** [comment]
Bump

**Comment by chergert** [comment]
Sorry I missed this. I don't get email notifications for PRs for some reason.

**Pull request merged**